servers/physics_2d/space_2d_sw.cpp

```diff
@@ -180,12 +180,15 @@ int Physics2DDirectSpaceStateSW::intersect_shape(const RID &p_shape, const Trans
 	Rect2 aabb = p_xform.xform(shape->get_aabb());
 	aabb = aabb.grow(p_margin);
 
-	int amount = space->broadphase->cull_aabb(aabb, space->intersection_query_results, p_result_max, space->intersection_query_subindex_results);
```

Comment:
why not doing MIN(p_result_max,Space2DSW::INTERSECTION_QUERY_MAX) ?

Reply:
If p_result_max is the lower value, it would lead to the same behavior as before, where the user input limit is applied to broad phase instead of narrow phase.